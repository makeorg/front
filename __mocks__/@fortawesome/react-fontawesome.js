
import React from 'react';

export const FontAwesomeIcon = ({ icon }) => (
  <svg
    lib="@fortawesome/react-fontawesome"
    icon={icon.iconName}
    prefix={icon.prefix}
  />
);
