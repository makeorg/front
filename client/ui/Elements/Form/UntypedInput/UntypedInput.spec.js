import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { faEnvelope } from '@fortawesome/free-regular-svg-icons';
import { UntypedInput } from './index';

describe('UntypedInput', () => {
  const defaultProps = {
    type: 'foo',
    name: 'bar',
    icon: faEnvelope,
    value: 'baz',
    label: 'qux',
    handleChange: () => {},
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer.create(<UntypedInput {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(
      <UntypedInput {...defaultProps} tabIndex="-1" />
    );
    const PositiveTabIndex = renderer.create(
      <UntypedInput {...defaultProps} tabIndex="0" />
    );
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when input is required or optionnal', () => {
    const RequiredInput = renderer.create(
      <UntypedInput {...defaultProps} required />
    );
    const OptionnalInput = renderer.create(
      <UntypedInput {...defaultProps} required={false} />
    );
    expect(snapshotDiff(RequiredInput, OptionnalInput)).toMatchSnapshot();
  });
});
