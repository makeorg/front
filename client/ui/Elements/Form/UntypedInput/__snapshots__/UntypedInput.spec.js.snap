// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UntypedInput must match the snapshot with default Props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 1.125rem;
  border-radius: 1.875rem;
  border: 0.0625rem solid rgba(0,0,0,0.2);
  margin-bottom: 0.9375rem;
  background-color: rgb(242,242,242);
  border-color: rgba(0,0,0,0.2);
}

.c2 {
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 0.625rem;
  width: 100%;
  color: rgba(0,0,0,0.55);
  font-size: 0.875rem;
  line-height: 2.188rem;
}

.c1 {
  width: 1.875rem;
  color: rgb(232,24,68);
  font-size: 1rem;
}

@media (min-width:47.94rem) {
  .c2 {
    font-size: 1rem;
    line-height: 2.5rem;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    font-size: 1.125rem;
  }
}

<div
  className="c0"
>
  <label
    aria-label="qux"
    className="c1"
    htmlFor="bar"
  >
    <svg
      icon="envelope"
      lib="@fortawesome/react-fontawesome"
      prefix="far"
    />
  </label>
  <input
    aria-required={false}
    className="c2"
    id="bar"
    name="bar"
    onChange={[Function]}
    placeholder="qux"
    required={false}
    tabIndex={0}
    type="foo"
    value="baz"
  />
</div>
`;

exports[`UntypedInput must return the diff between snapshot when input is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -11,17 +11,17 @@
        lib="@fortawesome/react-fontawesome"
        prefix="far"
      />
    </label>
    <input
-     aria-required={true}
+     aria-required={false}
      className="ntcd9w-0 ntcd9w-7 osRhJ"
      id="bar"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
-     required={true}
+     required={false}
      tabIndex={0}
      type="foo"
      value="baz"
    />
  </div>
`;

exports[`UntypedInput must return the diff between snapshot with a negative vs a postive Tab Index 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -18,10 +18,10 @@
      id="bar"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
-     tabIndex="-1"
+     tabIndex="0"
      type="foo"
      value="baz"
    />
  </div>
`;
